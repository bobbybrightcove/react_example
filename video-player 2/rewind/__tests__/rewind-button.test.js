
import React from 'react';
import renderer from 'react-test-renderer';

import RewindButton from '../rewind-button';


describe('RewindButton', () => {
  const player = {};

  beforeAll(() => {
    player.currentTime = jest.fn().mockImplementation(() => 40);
  });

  afterAll(() => jest.resetAllMocks());

  it('renders according to valid snapshot', () => {
    const component = renderer.create(
      <RewindButton
        player={player}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls player.currentTime twice', () => {
    const component = renderer.create(
      <RewindButton
        player={player}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(player.currentTime).toHaveBeenCalledTimes(2);
  });
});
