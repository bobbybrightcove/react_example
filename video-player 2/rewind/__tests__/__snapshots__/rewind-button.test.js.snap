// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewindButton calls player.currentTime twice 1`] = `
<img
  alt="Rewind"
  onClick={[Function]}
  src="rewind.svg"
/>
`;

exports[`RewindButton calls player.currentTime twice 2`] = `
<img
  alt="Rewind"
  onClick={[Function]}
  src="rewind.svg"
/>
`;

exports[`RewindButton renders according to valid snapshot 1`] = `
<img
  alt="Rewind"
  onClick={[Function]}
  src="rewind.svg"
/>
`;
